File: templates/java-webflux/subsystem/user/interface/src/main/java/cobalt/user/contract/UserWithPosts.java
package cobalt.user.contract;

import cobalt.post.model.Post;
import cobalt.user.model.User;

import java.util.List;

public class UserWithPosts {

  public static class Res {
    Integer id;
    String familyName;
    String givenName;
    List<Post> posts;

    Res(Integer id, String familyName, String givenName, List<Post> posts) {
      this.id = id;
      this.familyName = familyName;
      this.givenName = givenName;
      this.posts = posts;
    }

    public static Res from(User user, List<Post> posts) {
      return new Res(user.id, user.familyName, user.givenName, posts);
    }
  }

}

File: templates/kotlin-webmvc/subsystem/user/interface/src/main/kotlin/{{PACKAGE_NAME}}/user/contract/UserWithPosts.kt
package {{PACKAGE_NAME}}.user.contract

import {{PACKAGE_NAME}}.post.model.Post
import {{PACKAGE_NAME}}.user.model.User

sealed class UserWithPosts {
  data class Res(
    val id: Int,
    val familyName: String,
    val givenName: String,
    val posts: List<Post>
  ) {
    companion object {
      fun from(user: User, posts: List<Post>) =
        Res(
          id = user.id,
          familyName = user.familyName,
          givenName = user.givenName,
          posts = posts
        )
    }
  }
}
